src/drivers/mct/main/prep_glc_mod.F90

```diff
@@ -136,27 +138,32 @@ subroutine prep_glc_init(infodata, lnd_c2_glc)
 
     smb_renormalize = prep_glc_do_renormalize_smb(infodata)
 
+    if ((glc_present .and. lnd_c2_glc) .or. do_hist_l2x1yrg) then
+
+       l2x_lx => component_get_c2x_cx(lnd(1))
+       lsize_l = mct_aVect_lsize(l2x_lx)
+
+       allocate(l2gacc_lx(num_inst_lnd))
+       do eli = 1,num_inst_lnd
+          call mct_aVect_init(l2gacc_lx(eli), rList=seq_flds_l2x_fields_to_glc, lsize=lsize_l)
+          call mct_aVect_zero(l2gacc_lx(eli))
+       end do
+       l2gacc_lx_cnt = 0
```

Comment:
The code inside this conditional block is not new: I simply moved a subset of the code from the below conditional up here, into this new conditional which is executed under the additional condition of `do_hist_l2x1yrg` (in addition to the old condition of `glc_present .and. lnd_c2_glc`).